services/src/main/java/org/keycloak/authentication/authenticators/browser/CookieAuthenticator.java

```diff
@@ -83,6 +84,15 @@ public void close() {
 
     }
 
+    protected boolean requireReauthentication(UserSessionModel userSession, ClientSessionModel clientSession) {
+        return isPromptLogin(clientSession) || isAuthTimeExpired(userSession, clientSession);
+    }
+
+    protected boolean isPromptLogin(ClientSessionModel clientSession) {
```

Comment:
Yes. It's always the question, whether to go with something easy or with something more generic, but usually more complex ;)

The logic for prompt=login or "max_age" needs to be somewhere. For me, it look the easier and better option is to add into the Authenticator implementation itself rather then to the framework and increase the complexity of the framework.

Note that we also need to update (or don't update) auth_time based on SSO authentication happen or "real" authentication happen.

So how you expect the generic approach to work? I can see 2 possibilities:
- Have 2 authentication flows - one with cookie authenticator and one without cookie authenticator. The flow will be chosen according to "prompt=login" or "max_age=XX" is used. 
- Remove CookieAuthenticator and instead implement SSO cookie check before authenticator framework is triggered. The SSO cookie check will be skiped if you use "prompt=login" or "max_age=XX". If Cookie is valid, the Authentication framework won't be triggered at all and user will be considered authenticated.

TBH both options looks to me more complex then just add the behaviour directly into Cookie authenticator.